randovania/gui/lib/game_exporter.py

```diff
@@ -54,12 +52,12 @@ def work(progress_update: ProgressUpdateCallable):
 
     except UnableToExportError as e:
         logging.warning(e.reason)
-        progress_update_signal.emit(f"Unable to export game: {e.reason}", 0.0)
+        background.progress_update_signal.emit(f"Unable to export game: {e.reason}", 0.0)
```

Comment:
That error here is so confusing for me. This is because we cannot type a qt signal properly?

Reply:
yeah, `QtCore.Signal` just doesn't seem to have the `emit` field or the correct signature or be valid typing if you do the `Signal(str, float)`. It's weird.

Reply:
Alright. Then I press the merge button

Reply:
If I understand correctly, I figured that it actually comes from something different.
The type stub for `Signal` is defined as 
```python
    @typing.overload
    def __get__(self, instance: PySide6.QtCore.QObject, owner: typing.Any | None, /) -> PySide6.QtCore.SignalInstance: ...
    @typing.overload
    def __get__(self, instance: None, owner: typing.Any | None, /) -> PySide6.QtCore.Signal: ...
```
`instance` in `BackgroundTaskMixin` is neither `None` nor a `PySide6.QtCore.QObject`. 
Could be "fixed" by inherting from `PySide6.QtCore.QObject` but that will introduce much more trouble because then you get `RuntimeError: You can't initialize an PySide6.QtWidgets.QMainWindow object in class MainWindow twice!`

Reply:
incidentally, this happens to be the only case in the codebase of a `Signal` being used on something that isn't a `QObject`

Reply:
Might be the reason to implement our own `Signal`, I don't like the connectors being QObjects just for it.